### PR TITLE
Add github CI 

### PR DESCRIPTION
**Pull request by adambujak**

### CODE REVIEW

File: .github/workflows/build_check.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build the discovery firmware
+      run: make disco -C firmware
+    - name: Build the nucleo firmware
+      run: make disco -C firmware
```
Comment (stephen99scott) [inline .github/workflows/build_check.yml:20]: Typo

Comment (adambujak) [reply .github/workflows/build_check.yml:20]: thanks

Comment (adambujak) [reply .github/workflows/build_check.yml:20]: sorry this should be a draft
I'm experimenting rn

Comment (stephen99scott) [reply .github/workflows/build_check.yml:20]: Np thanks for setting this up

Comment (adambujak) [reply .github/workflows/build_check.yml]: np ready for review

File: firmware/tools/run_formatter.sh [hunk 2]
```diff
@@ -1,18 +1,78 @@
...
+{
+    docker rm -f env > /dev/null
+}
+
+runFormatter ()
+{
+    # get a list of all the files we want to run the formatter on
+    docker exec -i env bash -c "find /workspace/. \( -name '*.c' -o -name '*.h' -o -path '/workspace/./dependencies' -prune \) -type f > $TEMP_FIND_FILE"
+
+    if [ $CHECK_FLAG == "TRUE" ]; then
```
Comment (stephen99scott) [inline firmware/tools/run_formatter.sh:44]: I think this should be double brackets

File: .github/workflows/run_checks.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  format_check:
+    needs: build
+    runs-on: ubuntu-latest
+    if: always()
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check Code Formatting
+      run: make format_check -C firmware
```
Comment (stephen99scott) [inline .github/workflows/run_checks.yml:43]: What's the purpose of "firmware" here

Comment (adambujak) [reply .github/workflows/run_checks.yml:43]: to run it in the firmware folder

File: firmware/tools/run_formatter.sh (hunk 2 above)
Comment (adambujak) [reply firmware/tools/run_formatter.sh:44]: single brackets here are fine, but I'm fine with making double brackets our standard if that's what we want. It's less portable apparently but I don't think that's a huge issue here

File: .github/workflows/run_checks.yml (hunk 3 above)
Comment (adambujak) [reply .github/workflows/run_checks.yml:43]: more specifically, to run the make command in the firmware folder

Comment (adambujak) [reply .github/workflows/run_checks.yml:43]: https://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_node/make_93.html

File: firmware/tools/run_formatter.sh (hunk 2 above)
Comment (stephen99scott) [reply firmware/tools/run_formatter.sh:44]: There's an exception on this line when I run in mingw without double brackets

Comment (adambujak) [reply firmware/tools/run_formatter.sh:44]: oh really? that's odd. I'll change it

File: .github/workflows/run_checks.yml (hunk 3 above)
Comment (stephen99scott) [reply .github/workflows/run_checks.yml:43]: I see thanks, I was confusing -C with the -c passed to the run_formatter script

File: firmware/tools/run_formatter.sh (hunk 2 above)
Comment (stephen99scott) [reply firmware/tools/run_formatter.sh]: I had that issue before with mingw, I'm not sure why it's like that

**Review by stephen99scott — APPROVED** [review]

**Comment by stephen99scott** [comment]
@adambujak  If the format check fails, i.e. one of the files doesn't have the keyword "pass", will this check still pass on GitHub? It might make sense to add in some sort of parser that read's the output and scans for "fail" keywords

**Comment by adambujak** [comment]
> @adambujak If the format check fails, i.e. one of the files doesn't have the keyword "pass", will this check still pass on GitHub? It might make sense to add in some sort of parser that reads the output and scans for "fail" keywords

I'm almost certain if it fails for one of them the the check will fail on github. Did you see something to the contrary?

**Comment by stephen99scott** [comment]
> I'm almost certain if it fails for one of them the the check will fail on github. Did you see something to the contrary?

I was just making sure, I wasn't sure if uncrustify would return an error or not, but it looks like it does so that's good. Nice

**Pull request merged**